src/action/macos/create_nix_volume.rs

```diff
@@ -71,39 +73,55 @@ impl CreateNixVolume {
             .map_err(Self::error)?;
 
         let encrypt_volume = if encrypt {
-            Some(EncryptApfsVolume::plan(disk, &name, &create_volume).await?)
+            Some(EncryptApfsVolume::plan(nix_enterprise, disk, &name, &create_volume).await?)
         } else {
             None
         };
 
-        let setup_volume_daemon = CreateVolumeService::plan(
-            NIX_VOLUME_MOUNTD_DEST,
-            "org.nixos.darwin-store",
-            name.clone(),
-            "/nix",
-            encrypt,
-        )
-        .await
-        .map_err(Self::error)?;
+        let setup_volume_daemon: Option<StatefulAction<CreateVolumeService>>;
+        let bootstrap_volume: Option<StatefulAction<BootstrapLaunchctlService>>;
+        let kickstart_launchctl_service: Option<StatefulAction<KickstartLaunchctlService>>;
 
-        let bootstrap_volume = BootstrapLaunchctlService::plan(
-            "system",
-            "org.nixos.darwin-store",
-            NIX_VOLUME_MOUNTD_DEST,
-        )
-        .await
-        .map_err(Self::error)?;
-        let kickstart_launchctl_service =
-            KickstartLaunchctlService::plan("system", "org.nixos.darwin-store")
+        if !nix_enterprise {
```

Comment:
Since there's an `else` branch, I'd argue that this should be `if nix_enterprise { ... = None; }` first. It feels weird to me to have `if false ... else if true ...` instead of `if true ... else if false ...`.